pennylane/transforms/core/transform_dispatcher.py

```diff
@@ -92,6 +96,54 @@ def is_informative(self):
         """Return True is the transform does not need to be executed."""
         return self._is_informative
 
+    def custom_qnode_transform(self, fn):
```

Comment:
Maybe we could just name this `register_custom_qnode_transform`, eliminate a public `default_qnode_transform`?

Reply:
I think I prefer your first solution with adding a `qnode_transform` keyword argument to the initialization. This prevents changing the transform at runtime.